src/StateService/Storage/StateStore.cs

```diff
@@ -54,11 +54,11 @@ public StateSnapshot GetSnapshot()
             return new StateSnapshot(store);
         }
 
-        public HashSet<byte[]> GetProof(UInt256 root, StorageKey skey)
+        public bool TryGetProof(UInt256 root, StorageKey skey, out HashSet<byte[]> proof)
         {
             using ISnapshot snapshot = store.GetSnapshot();
             var trie = new MPTTrie<StorageKey, StorageItem>(snapshot, root);
-            return trie.GetProof(skey);
+            return trie.TryGetProof(skey, out proof);
         }
 
         protected override void OnReceive(object message)
```

Comment:
On `UpdateLocalStateRoot` we don't check the result of `Trie.Put`

Reply:
I will remove the bool result. Put is always expected to be successful or throwing exception.